virttest/qemu_vm.py

```diff
@@ -1192,12 +1192,15 @@ def add_boot(devices, boot_order, boot_once, boot_menu, boot_strict):
                 logging.warn("-boot on s390x only support boot strict=on")
                 return "-boot strict=on"
             cmd = " -boot"
-            patterns = ["order", "once", "menu", "strict"]
+            patterns = ["order", "once", "menu", "strict", "reboot-timeout"]
+            boot_reboot_timeout = params.get('boot_reboot-timeout', 60)
             options = []
             for p in patterns:
                 pattern = "boot .*?(\[,?%s=(.*?)\]|\s+)" % p
                 if devices.has_option(pattern):
+                    p = p.replace("-", "_")
                     option = locals()["boot_%s" % p]
+                    p = p.replace("_", "-")
```

Comment:
Suggest changing the interface to `def add_boot(device, args):`, and modify the caller to:
```
        if devices.has_option("boot"):
            boot_args = {}
            boot_args["order"] = params.get("boot_order", "cdn")
            boot_args["once"] = params.get("boot_once", "c")
            ...
            boot_args["reboot-timeout"] = params.get("boot_reboot_timeout")
            cmd = add_boot(devices, boot_args)
```
Then, you do not have to make this conversion.

Reply:
@luckyh Thanks Xu,  I have updated it, please review it again.